app/Converters/parser_onnx.py

```diff
@@ -31,12 +33,29 @@ def onnx_to_json(model_path, output_json_path):
     }
 
     layer_info = []
+
+    input_info = {}
+    for input in model.graph.input:
+        if input.name in initializers_dict:
+            continue
+
+        input_info = {
+            "name": input.name,
+            "shape": [dim.dim_value for dim in input.type.tensor_type.shape.dim],
```

Comment:
`dim.dim_value` is 0 for dynamic dims. Consider handling `dim_param` or mapping 0→-1/None to preserve dynamic shape semantics in JSON.